packages/shared-ux/markdown/impl/markdown.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiMarkdownEditor, EuiMarkdownEditorProps, EuiMarkdownFormat } from '@elastic/eui';
+import React, { useState } from 'react';
+
+export type MarkdownProps = Partial<
+  Omit<EuiMarkdownEditorProps, 'editorId' | 'uiPlugins' | 'markdownFormatProps'>
+> & {
+  /**
+   * @param readOnly is needed to differentiate where markdown is used as a presentation of error messages
+   * This was previous the MarkdownSimple component
+   */
+  readOnly: boolean;
+  defaultValue?: string;
+  markdownContent?: string;
+  ariaLabelContent?: string;
+  /** Eui allows the height of the markdown component to be set */
+  height?: number | 'full';
+  placeholder?: string | undefined;
+  children?: string;
+};
+
+export const Markdown = ({
+  ariaLabelContent,
+  readOnly,
+  markdownContent,
+  children,
+  defaultValue = '',
+  placeholder = '',
+  height = 'full',
+}: MarkdownProps) => {
+  const [value, setValue] = useState(defaultValue);
+
+  // Render EuiMarkdownFormat
+  if (readOnly && children) {
```

Comment:
nit: you can simplify the JSX with

```ts
if (readOnly) {
  return (
    <EuiMarkdownFormat aria-label={ariaLabelContent ?? 'markdown component'}>
      {children ?? markdownContent}
    </EuiMarkdownFormat>
  );
}

// Otherwise render the Markdown Editor
return (
  <EuiMarkdownEditor
    readOnly={readOnly}
    aria-label={ariaLabelContent ?? 'markdown component'}
    placeholder={placeholder}
    value={value}
    onChange={setValue}
    height={height}
  />
);
```

Reply:
Edit: you probably want to `throw` on readonly if no content is passed

```ts
if (readOnly) {
  if (!children && !markdownContent) {
    throw new Error('Markdown content is required in [readOnly] mode.');
  }
  return (
    <EuiMarkdownFormat aria-label={ariaLabelContent ?? 'markdown component'}>
      {children ?? markdownContent}
    </EuiMarkdownFormat>
  );
}
```

Reply:
Nice thanks!! I added a separate story since the error handling will effectively error on storybook